File: printerservice/src/main/java/com/example/print/printerserver/java_connectors/StreamPrinterConnector.java
package com.example.print.printerserver.java_connectors;

import java.io.DataOutputStream;
import java.io.IOException;
import java.io.InputStream;

public class StreamPrinterConnector extends PrinterConnector {

    private InputStream inputStream;

    public StreamPrinterConnector(InputStream inputStream) {
        this.inputStream = inputStream;
    }

    @Override
    public void writeData(DataOutputStream out) throws IOException {
        byte[] buffer = new byte[3000];
        while (inputStream.read(buffer) != -1)
            out.write(buffer);
    }
}

File: printerservice/src/main/java/com/example/print/printerserver/connectors/StreamPrinterConnector.kt
package com.example.print.printerserver.connectors

import com.example.print.printerserver.utils.write
import java.io.DataOutputStream
import java.io.InputStream

internal class StreamPrinterConnector(private val data: InputStream) : PrinterConnector() {

    override fun writeData(output: DataOutputStream?) {
        output?.write(data)
    }
}